### DISCUSSION TITLE
Design pattern in MSA

### DISCUSSION THREAD

**Discussion by sonck201** [nashtech-garage/nodejs-msa#34] [Ideas]
Which microservices design patterns should be implemented this the framework also?

**Comment by sonck201** [comment]
```mermaid
flowchart LR
  FE --> BFF --> BE --> DB
```

**Reply by sonck201** [reply]
```mermaid
flowchart LR
  subgraph microService["Micro service"]
    BE -- Query --> postgres
  end
```

**Comment by panoti** [comment]
- [ ] BFF
- [ ] Circuit breaker
- [ ] Request/Reply
- [ ] Event sourcing
- [ ] Database per Microservice
- [ ] 2PC/Outbox/Saga
- [ ] Service mesh

**Comment by VuiLenDi** [comment]
Hi @tuonglevan, do we continue with these ideas? or do we need update anything?

**Reply by tuonglevan** [reply]
Hi a @VuiLenDi these ideas all good to me.
